tests/models/test_taskinstance.py

```diff
@@ -551,15 +551,15 @@ def task_function(ti):
         assert ti.state == State.UP_FOR_RETRY
 
     @pytest.mark.skip_if_database_isolation_mode  # Does not work in db isolation mode as DB access in code
-    @pytest.mark.parametrize("state", [State.SUCCESS, State.FAILED, State.SKIPPED])
-    def test_task_sigterm_doesnt_change_state_of_finished_tasks(self, state, dag_maker):
+    @pytest.mark.parametrize("state", [State.SUCCESS, State.FAILED, State.SKIPPED, None])
+    def test_task_sigterm_doesnt_change_state_of_finished_and_cleared_tasks(self, state, dag_maker):
         session = settings.Session()
 
         def task_function(ti):
             ti.state = state
             session.merge(ti)
             session.commit()
-            raise AirflowException()
+            raise AirflowTaskTerminated()
```

Comment:
Changed to dedicated exception introduced with https://github.com/apache/airflow/pull/37613